torch/csrc/jit/interpreter.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+#include <memory>
+#include <vector>
+
+namespace at {
+  struct Tensor;
+}
+namespace torch { namespace jit {
+
+struct NotImplementedException : public std::logic_error {
+  NotImplementedException()
+  : std::logic_error("Function not yet implemented.") {}
+};
+
+// The interpreter run Graphs with Tensor inputs and Tensor outputs
+// a separate component in the autograd handles unwrapping and wrapping
+// variable objects for use in the interpreter.
+
+struct FunctionImpl;
+struct InterpreterImpl;
+struct Graph;
+
+struct Function {
```

Comment:
Blah, this is the same name as autograd Function, which will make it harder to grep for instances of this particular concept. How about `GraphBCO` (graph bytecode object).

Reply:
(Yes, I know we don't actually have a bytecode today. But we might, someday...)

Reply:
I did Function -> Code, and Interpreter -> InterpreterState.